src/main/java/gregtech/loaders/recipe/RecyclingRecipes.java

```diff
@@ -1,111 +1,400 @@
+        // assume it can be anywhere in the list.
+        MaterialStack itemMs = materials.stream().filter(ms -> !ms.material.equals(fluidMs.material)).findFirst().orElse(null);
+
+        // Calculate the duration based off of those two possible outputs.
+        // - Sum the two Material amounts together (if both exist)
+        // - Divide the sum by M
+        long duration = fluidMs.amount * fluidMs.material.getAverageMass();
+        if (itemMs != null) duration += (itemMs.amount * itemMs.material.getAverageMass());
+        duration = Math.max(1L, duration / M);
+
+        // Build the final Recipe.
+        RecipeBuilder<?> extractorBuilder = RecipeMaps.EXTRACTOR_RECIPES.recipeBuilder()
+                .inputs(input)
+                .fluidOutputs(fluidMs.material.getFluid((int) (fluidMs.amount * L / M)))
+                .duration((int) duration)
+                .EUt(32 * multiplier);
+
+        // Null check the Item before adding it to the Builder.
+        // - Try to output an Ingot, otherwise output a Dust.
+        if (itemMs != null) {
+            OrePrefix prefix = itemMs.material.hasProperty(PropertyKey.INGOT) ? OrePrefix.ingot : OrePrefix.dust;
+            extractorBuilder.output(prefix, itemMs.material, (int) (itemMs.amount / M));
         }
+
+        extractorBuilder.buildAndRegister();
+    }
+
+    private static void registerArcRecycling(ItemStack input, List<MaterialStack> materials, int multiplier) {
+
+        // Filter down the materials list.
+        // - Map to the Arc Smelting result as defined below
+        // - Combine any MaterialStacks that have the same Material
+        materials = combineStacks(materials.stream()
+                .map(RecyclingRecipes::getArcSmeltingResult)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList()));
+
+        // Finalize the output List
+        List<ItemStack> outputs = finalizeOutputs(
+                materials,
+                RecipeMaps.ARC_FURNACE_RECIPES.getMaxOutputs(),
+                RecyclingRecipes::getArcIngotOrDust
+        );
+
+        // Exit if no valid outputs exist for this recycling Recipe.
+        if (outputs.size() == 0) return;
+
+        // Build the final Recipe.
+        RecipeMaps.ARC_FURNACE_RECIPES.recipeBuilder()
+                .inputs(input)
+                .outputs(outputs)
+                .duration(calculateDuration(materials))
+                .EUt(30 * multiplier)
+                .buildAndRegister();
     }
 
-    private static ItemStack getArcSmeltingResult(MaterialStack materialStack) {
+    private static MaterialStack getArcSmeltingResult(MaterialStack materialStack) {
         Material material = materialStack.material;
-        long materialAmount = materialStack.amount;
+        long amount = materialStack.amount;
+
+        if (material.hasFlag(EXPLOSIVE)) {
+            return new MaterialStack(Materials.Ash, amount / 16);
```

Comment:
Should these return at minimum 1, to ensure that all the inputs would be accounted for, or are you find returning a materialStack with size 0 if `amount` is less than your divisor?

Reply:
The material stack amount after dividing should never reasonably be zero here, at worst it would be something like `OrePrefix.dustTiny.materialAmount / 16`, or `403200 / 16` = 25,200 (this would still not create any output, but that is fine in a lot of cases)